src/analyses/base.ml

```diff
@@ -1898,7 +1894,38 @@ struct
         ) else
           M.warn_each ~ctx:ctx.control_context msg
     in
-    match check_assert e ctx.local with
+    let meet_results a b = 
+        match (a, b) with
+        | (a, `Bot) -> `Bot
+        | (`Bot, b) -> `Bot
+        | (a, `Top) -> a
+        | (`Top, b) -> b
+        | (`True, `False) -> `Bot
+        | (`False, `True) -> `Bot
+        | (a, b) ->  a
```

Comment:
This seems quite annoying. I think we already have a flat boolean lattice somewhere (`Queries.AR` seems to be what you added) that has this operation. I guess the issue here was the special `True` and `False` instead of `Lifted true` and `Lifted false`. It might make more sense to switch from the former to the latter, so we don't have to duplicate the logic and convert between them below.